app/models/user.rb

```diff
@@ -1,8 +1,12 @@
 class User < ActiveRecord::Base
   authenticates_with_sorcery!
 
-  validates :email, :password, presence: true
+  validates :email, presence: true
   validates :email, uniqueness: true, if: -> { email.present? }
+  validates :password, presence: true, on: :update, confirmation: true
+
+  before_create :setup_activation
```

Comment:
Sorcery provides this it will set the following attributes `activation_state`, `activation_token` and `activation_token_expires_at`.